### ISSUE TITLE
Problem for nodejs 0.1.98

### ISSUE THREAD

**Issue by lsm** [akaspin/nun#3]
```
undefined:3
var HS = new HStream();
^
ReferenceError: HStream is not defined
    at eval at compile (/path/to/nun/lib/nun/compiler.js:58:2)
    at /path/to/nun/lib/nun/index.js:73:23
    at /path/to/nun/lib/nun/index.js:40:6
    at Object.get (/path/to/nun/lib/nun/cache.js:43:15)
    at /path/to/nun/lib/nun/index.js:38:10
    at EventEmitter._tickCallback (node.js:48:25)
    at node.js:221:9
```

**Comment by akaspin** [comment]
Please give me your code that produces this error. I can't reproduce it - all tests passed.

**Comment by lsm** [comment]
Hi, 
here's the code and how you can reproduce:

```
var nun = require("./lib/nun"),
path = require('path'),
http = require('http');

http.createServer(function(request, response) {
    nun.render(path.join(__dirname, '/sample.html'), {}, {}, function (err, output) {
        if (err) {
            throw err;
        }
        var buffer = '';
        output.addListener('data', function (c) {
            buffer += c;
        })
        .addListener('end', function () {
            response.writeHead(200, {'Content-Type': 'text/html', 'Content-Length': buffer.length});
            response.write(buffer, 'utf8');
            response.end();
        });
    });
}).listen(8000, '127.0.0.1');
```

content of sample.html: 
    <html></html>

save the about code and simply run:
    node yourScriptName.js
then open http://127.0.0.1:8000/, 
the first time you'll get a correct response
wait for about 10 seconds and refresh you browser the above exception will throw.

not sure it's a nodejs bug or not, working without problem with node 0.1.97

**Comment by akaspin** [comment]
Ok. It' not node.js bug. It's V8 bug. Sad but true. At http://github.com/ry/node/commit/cd232a9ccecb5102170d4d40c91f06bb538104e0 V8 is upgraded to 2.2.16.

I filed bug http://github.com/ry/node/issues/#issue/175. Can you reproduce it?

**Comment by lsm** [comment]
Yes, I got the same output as you posted in #[175](http://github.com/ry/node/issues/#issue/175). 
While, if the reference to the required object registered in global, everything goes as usual:

in a.js change the first two line into:
    sys = require("sys");
    b_op = require('./b').op;

as you said, maybe it's some kind of GC introduced in V8 2.2.6, the [change log](http://code.google.com/p/v8/source/browse/trunk/ChangeLog) may helps

**Comment by akaspin** [comment]
Yes it's solves one problem. But I think that register in global is way to hell. Another one: after 10 secs we lost any closures in eval. So I start digging `Script`

**Comment by akaspin** [comment]
Closed by 552db0c56b26603dd67f33702972da6ba8b379a7

**Comment by lsm** [comment]
[Script](http://nodejs.org/api.html#script-91) rocks;)
